nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/AsyncService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nelson Silva <nsilva@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.api;
+
+import java.io.Serializable;
+
+/**
+ * Interface to be implemented by asynchronous services.
+ *
+ * @param <K> type of command id
+ * @param <S> type of status
+ * @param <V> type of result
+ * @since 10.3
+ */
+public interface AsyncService<K extends Serializable, S, V> {
```

Comment:
should not S extends AsyncStatus ?

Reply:
I actually had it at some point then removed. TBH just kept `getStatus()` so that I can return it in the status endpoint and because we return it today when calling `abort()`. I don't really think an `AsyncService` needs to expose a status given we can have `isCompleted()` and friends...

Reply:
👍